persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideActor.scala

```diff
@@ -3,103 +3,108 @@
-import scala.concurrent.duration.FiniteDuration
 
 private[lagom] object ReadSideActor {
 
   def props[Event <: AggregateEvent[Event]](
-    processor:            () => ReadSideProcessor[Event],
-    eventStreamFactory:   (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
-    clazz:                Class[Event],
-    globalPrepareTask:    ClusterStartupTask,
-    globalPrepareTimeout: FiniteDuration
+    config:             ReadSideConfig,
+    clazz:              Class[Event],
+    globalPrepareTask:  ClusterStartupTask,
+    eventStreamFactory: (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
+    processor:          () => ReadSideProcessor[Event]
   )(implicit mat: Materializer) = {
-    Props(classOf[ReadSideActor[Event]], processor, eventStreamFactory, clazz, globalPrepareTask, globalPrepareTimeout, mat)
+    Props(
+      classOf[ReadSideActor[Event]],
+      config,
+      clazz,
+      globalPrepareTask,
+      eventStreamFactory,
+      processor,
+      mat
+    )
   }
 
-  /**
-   * Start processing from the given offset
-   */
-  case class Start(offset: Offset)
+  case object Start
+
 }
 
 /**
  * Read side actor
  */
 private[lagom] class ReadSideActor[Event <: AggregateEvent[Event]](
-  processorFactory:     () => ReadSideProcessor[Event],
-  eventStreamFactory:   (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
-  clazz:                Class[Event],
-  globalPrepareTask:    ClusterStartupTask,
-  globalPrepareTimeout: FiniteDuration
+  config:             ReadSideConfig,
+  clazz:              Class[Event],
+  globalPrepareTask:  ClusterStartupTask,
+  eventStreamFactory: (AggregateEventTag[Event], Offset) => Source[akka.japi.Pair[Event, Offset], NotUsed],
+  processorFactory:   () => ReadSideProcessor[Event]
 )(implicit mat: Materializer) extends Actor with ActorLogging {
+
   import ReadSideActor._
   import akka.pattern.pipe
-  import akka.pattern.ask
   import context.dispatcher
 
   private var shutdown: Option[KillSwitch] = None
 
+  override def preStart(): Unit = {
+    super.preStart()
+  }
+
   override def postStop: Unit = {
     shutdown.foreach(_.shutdown())
   }
 
   def receive = {
     case EnsureActive(tagName) =>
-
-      val tag = AggregateEventTag.of(clazz, tagName)
-
-      implicit val timeout = Timeout(globalPrepareTimeout)
-
-      globalPrepareTask.askExecute() pipeTo self
-      context become preparing(tag)
-  }
-
-  def preparing(tag: AggregateEventTag[Event]): Receive = {
-
-    case Done =>
-      val handler = processorFactory().buildHandler()
-      handler.prepare(tag).toScala.map(Start) pipeTo self
-      context become active(handler, tag)
-
-    case Status.Failure(e) =>
-      throw e
-
+      implicit val timeout = Timeout(config.globalPrepareTimeout)
+      globalPrepareTask.askExecute().map { _ => Start } pipeTo self
+      context.become(start(tagName))
```

Comment:
This is identical but instead of piping a `Done` it pipes a `Start`.

The number of possible behaviors goes from 3 to 2 because the `preparing` and the `active` behaviours become `start`.